app/javascript/components/upload/StepTabHeader.js

```diff
@@ -38,14 +38,14 @@ export default function StepTabHeader({
     badgeContent = <span className="badge fa-xs" data-testid={badgeTestId}><FontAwesomeIcon icon={faCircle}/></span>
   }
 
-  return <li className={className}>
-    <div onClick={() => setCurrentStep(step)} className="step-number">
-      {badgeContent}
-    </div>
-    <div>
-      <a className="action link" onClick={() => setCurrentStep(step)}>
+  return <li className={className} role="tab">
+    <button className="unset-background-border-style" onClick={() => setCurrentStep(step)} >
```

Comment:
This is a good upgrade to the markup, we just need to update our tab-detection in metrics-api.js to also catch buttons that are under tablists.  So I would extract the logic in lines 161-167 in metrics-api.js into a helper function, and then make sure that helper function is called from both `logClickLink` and `logClickButton`

Reply:
I think I've addressed this as you described in the last commit, please let me know if that fits what you were thinking!